File: Amazon_LeetCode_Python_C++/ArrayString/1_two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++) {
            if (m.find(target - nums.at(i)) != m.end()) {
                ans.push_back(i);
                ans.push_back(m[target - nums.at(i)]);
            }
            m[nums.at(i)] = i;
        }
        return ans;
    }
};

File: Amazon_LeetCode_Python_C++/ArrayString/1_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = dict()
        ans = list()
        for i, n in enumerate(nums):
            if target - n in mp:
                ans.append(i)
                ans.append(mp[target - n])
            mp[n] = i
                
        return ans
